lib/tokenlist.cpp

```diff
@@ -796,8 +796,19 @@ static void compileTerm(Token *&tok, AST_state& state)
             }
         } else if (!state.cpp || !Token::Match(tok, "new|delete %name%|*|&|::|(|[")) {
             tok = skipDecl(tok);
-            while (tok->next() && tok->next()->isName())
-                tok = tok->next();
+            bool repeat = true;
+            while (repeat)
+            {
+                repeat = false;
+                if (tok->next() && tok->next()->isName()) {
+                    tok = tok->next();
+                    repeat = true;
+                }
+                if (Token::simpleMatch(tok->next(), "<") && tok->next()->link() && tok->next()->link()->next() && tok->next()->link()->next()->isName()) {
```

Comment:
you can write:

    if (Token::simpleMatch(tok->next(), "<") && Token::Match(tok->linkAt(1), "> %name%"))

In theory it would be enough to write:

    if (tok->next() && Token::Match(tok->linkAt(1), "> %name%"))

But well the simpleMatch makes it a bit clearer imho.